### PR TITLE
[RELAX] Adds a table of all nodes and whether they were tested, k and branch length

### PR DESCRIPTION
**Pull request by rdvelazquez**
Connected to #90.

Still to do:
- [x] Add sorting on columns
- [x] Format numbers (decimal places)
- [x] Add descriptions of column headers when hovered over
- [x] Confirm that the table works for both minimal (no k) and full analyses
- [x] Review code style (variable names, structure, comments)

@stephenshank feel free to review now or wait until the PR is completed (I'll ping you again when it is).

P.S. The git diff is also including trailing whitespace that was removed by my editor (atom's whitespace package). I've disabled this package so it won't happen in the future.

### CODE REVIEW

File: src/jsx/relax.jsx [hunk 1]
```diff
@@ -205,16 +206,43 @@ class RELAX extends React.Component{
...
+    var branchAttributes = data["branch attributes"][0];
+    var branchTestedStatuses = data["tested"][0];
+    var branchLengthsGTR = data["trees"]["branchLengths"]["Nucleotide GTR"];
+    // Step 2: Combine the info into one object.
+    var branchAttributesCombined = {};
+    for (var key in branchAttributes){
+      branchAttributesCombined[key] = {"Branch name": key, "Partition": branchTestedStatuses[key], "Branch length": branchLengthsGTR[key]};
+    }
+    // Step 3: Add "k" if it exists (i.e. the analysis was run as "all" vs. "minimal").
+    if(branchAttributes[_.keys(branchAttributes)[0]]["k (general descriptive)"]) {
```
Comment (stephenshank) [inline src/jsx/relax.jsx]: Try checking if key exists in the dictionary via something link

```"k (general descriptive)" in branchAttributes[_.keys(branchAttributes)[0]]```

File: src/jsx/relax.jsx [hunk 2]
```diff
@@ -205,16 +206,43 @@ class RELAX extends React.Component{
...
+      These arrays will be consumed by the DatamonkeyTable component.
+    */
+    // Step 1: Get individual branch information from JSON sources.
+    var branchAttributes = data["branch attributes"][0];
+    var branchTestedStatuses = data["tested"][0];
+    var branchLengthsGTR = data["trees"]["branchLengths"]["Nucleotide GTR"];
+    // Step 2: Combine the info into one object.
+    var branchAttributesCombined = {};
+    for (var key in branchAttributes){
+      branchAttributesCombined[key] = {"Branch name": key, "Partition": branchTestedStatuses[key], "Branch length": branchLengthsGTR[key]};
```
Comment (stephenshank) [inline src/jsx/relax.jsx]: For some reason, this is throwing an error on the CD2 and Flu datasets.

Comment (rdvelazquez) [reply src/jsx/relax.jsx]: It looks like neither of these files has a "trees" object. The data sets look like they were uploaded in October 2017 and the HyPhy JSON fields were updated in December 2017 (http://hyphy.org/resources/json-fields.pdf). I ran a test example on Datamonkey.org (http://datamonkey.org/relax/5a820007e7b8b163187b7005) and the JSON does have a trees file. 

Let me know your thoughts but I think this PR should be good to review now and we should open a new issue to update the data that comes with HyPhy-Vision to match the new JSON format.

Comment (rdvelazquez) [reply src/jsx/relax.jsx]: @stephenshank: As discussed, I re-ran the CD2 and Flu datasets on datamonkey.org and used the results to replace the existing JSON files in methods/RELAX/data.

**Comment by stephenshank** [comment]
@rdvelazquez There's still a strange bug when you switch from a JSON that contains results from a minimal vs. full analysis, where the `k` column heading goes missing (see attached .gif). Can you confirm this? I think we can still merge this if it's not an easy fix... just file it as an issue for later.

I would also consider the following:

1. Rename the column header "Partition" to "Branch partition" (other methods partition by sites, so I think it helps to be explicit).
2. Change the tooltip for the `k` column to read "Branch specific relaxation parameter".
3. Create a single `d3.format(".3r")` object that each entry references, as opposed to creating one per entry.

Other than that, I think this looks good... nice work!

relax-table-bug

**Pull request merged**